dashboard/app/models/user.rb

```diff
@@ -1159,26 +1159,62 @@ def in_progress_and_completed_scripts
     end
   end
 
-  # Return a collection of courses and scripts for the user. First in the list will
-  # be courses enrolled in by the user's sections. Following that will be all scripts
-  # in which the user has made progress that are not in any of the enrolled courses.
-  def recent_courses_and_scripts(exclude_primary_script)
-    courses = section_courses
-    course_scripts_script_ids = courses.map(&:default_course_scripts).flatten.pluck(:script_id).uniq
-
-    # filter out those that are already covered by a course
-    user_scripts = in_progress_and_completed_scripts.
-      select {|user_script| !course_scripts_script_ids.include?(user_script.script_id)}
-
-    course_data = courses.map do |course|
+  def assigned_courses
+    assigned_courses_data = section_courses.map do |course|
       {
         name: course[:name],
         title: data_t_suffix('course.name', course[:name], 'title'),
         description: data_t_suffix('course.name', course[:name], 'description_short'),
         link: course_path(course),
       }
     end
+    assigned_courses_data
+  end
+
+  def assigned_scripts(exclude_hidden)
+    assigned_user_scripts = user_scripts.where("assigned_at")
+
+    # get the scripts associated with the assigned_user_scripts
+    assigned_scripts = []
+    assigned_user_scripts.each do |user_script|
+      assigned_scripts << Script.where(id: user_script.script_id)
+    end
+
+    # filter out scripts that are already covered by an assigned course
+    course_scripts_script_ids = section_courses.map(&:default_course_scripts).flatten.pluck(:script_id).uniq
 
+    assigned_scripts = assigned_scripts.flatten.
+      select {|assigned_script| !course_scripts_script_ids.include?(assigned_script.id)}
+
+    # exclude hidden scripts if necessary
+    visible_assigned_scripts = []
+    assigned_scripts.each do |script|
+      unless script_hidden?(script)
+        visible_assigned_scripts << script
+      end
+    end
+
+    scripts = exclude_hidden ? visible_assigned_scripts : assigned_scripts
+
+    assigned_scripts_data = scripts.map do |script|
+      {
+        name: script[:name],
+        title: data_t_suffix('script.name', script[:name], 'title'),
+        description: data_t_suffix('script.name', script[:name], 'description_short', default: ''),
+        link: script_path(script)
```

Comment:
This block is the same as elsewhere in the file, and I suspect we depend on the two returning identical contents. It would be good to share this code.